Update: Write Read Status, Patch & List APIService test +3 endpoints

riaankleinhans opened the pull request:
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it:**
This PR adds a test to test the following untested endpoints:
- readApiregistrationV1APIServiceStatus
- patchApiregistrationV1APIService
- listApiregistrationV1APIService


**Which issue(s) this PR fixes:**
Fixes #97271

**Testgrid Link:**

**Special notes for your reviewer:**
Adds +3 endpoint test coverage (good for conformance)
This PR will replace #97272 by integrating the test with an existing test to reduce overhead.

**Does this PR introduce a user-facing change?:**
```
NONE

```

**Release note:**
```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:**
```
NONE

```

/sig testing
/sig architecture
/area conformance

Riaankl commented:
/triage accepted

johnbelamaric commented:
/retest

Since we are modifying an existing conformance test, we will also need to do the flake-checking by manually re-executing tests against the PR (since we can't merge and verify non-flakiness without possibly merging a flaky conformance test).

Riaankl commented:
/retest

Riaankl commented:
/test

Riaankl commented:
/test all

Riaankl commented:
```
I1216 19:01:40.042] Dec 16 19:01:35.751: INFO: Waiting for 30 pods to be running on the node; 20 are currently running;
I1216 19:01:40.042] Dec 16 19:01:35.752: FAIL: Unexpected error:
I1216 19:01:40.043]     <*errors.errorString | 0xc000192200>: {
I1216 19:01:40.043]         s: "timed out waiting for the condition",
I1216 19:01:40.043]     }
I1216 19:01:40.043]     timed out waiting for the condition
I1216 19:01:40.043] occurred
I1216 19:01:40.043] 
I1216 19:01:40.043] Full Stack Trace
I1216 19:01:40.043] k8s.io/kubernetes/test/e2e/node.glob..func4.2.3()
I1216 19:01:40.044] 	test/e2e/node/kubelet.go:360 +0x565
```
/retest

Riaankl commented:
/retest

Riaankl commented:
/retest

Riaankl commented:
/test pull-kubernetes-e2e-gce-ubuntu-containerd
/test pull-kubernetes-e2e-kind

Riaankl commented:
/test all

Riaankl commented:
/test all

Riaankl commented:
/test pull-kubernetes-conformance-image-test

Riaankl commented:
/test all

Riaankl commented:
/test all

Riaankl commented:
/test all

Riaankl commented:
6 runs of all jobs with no flake.
[Looking good](https://prow.k8s.io/pr-history/?org=kubernetes&repo=kubernetes&pr=97327)

johnbelamaric commented:
/approve for conformance

will need API machinery approval too

Riaankl commented:
/sig api-machinery

Riaankl commented:
/cc @deads2k @fedebongio 
The PR is an update to a current conformance test to add 3 more endpoints to conformance.
We have 6 clear runs of `/test all` to be sure it is safe to merge.
[PR History](https://prow.k8s.io/pr-history/?org=kubernetes&repo=kubernetes&pr=97327
) 
Just need /lgtm

Commit:
Extend APIService e2e testing of endpoints

e2e test validates the following 3 extra endpoints
- readApiregistrationV1APIServiceStatus
- patchApiregistrationV1APIService
- listApiregistrationV1APIService

Jefftree commented:
/lgtm

Riaankl commented:
Updated. Thank you for the constructive feedback @Jefftree

fedebongio commented:
/assign @lavalamp @deads2k 
Looks like the author wants to merge it sooner to get some soak time 
/triage accepted

Riaankl commented:
```
I1217 21:01:06.272] • Failure [48.567 seconds]
I1217 21:01:06.272] [sig-api-machinery] ResourceQuota
I1217 21:01:06.272] test/e2e/apimachinery/framework.go:23
I1217 21:01:06.272]   should create a ResourceQuota and capture the life of a service. [Conformance] [It]
I1217 21:01:06.272]   test/e2e/framework/framework.go:640
I1217 21:01:06.273] 
I1217 21:01:06.273]   Dec 17 21:00:56.312: Unexpected error:
      
I1217 21:01:06.273]       <*errors.errorString | 0xc000198200>: {
I1217 21:01:06.273]           s: "timed out waiting for the condition",
I1217 21:01:06.273]       }
I1217 21:01:06.273]       timed out waiting for the condition
I1217 21:01:06.273]   occurred
I1217 21:01:06.274] 
I1217 21:01:06.274]   test/e2e/apimachinery/resource_quota.go:112
I1217 21:01:06.274] ------------------------------
I1217 21:01:06.274] {"msg":"FAILED [sig-api-machinery] ResourceQuota should create a ResourceQuota and capture the life of a service. [Conformance]","total":-1,"completed":6,"skipped":26,"failed":1,"failures":["[sig-api-machinery] ResourceQuota should create a ResourceQuota and capture the life of a service. [Conformance]"]}
skipped 72079 lines unfold_more
I1217 21:17:53.595] 
I1217 21:17:53.596] [Fail] [sig-api-machinery] ResourceQuota [It] should create a ResourceQuota and capture the life of a service. [Conformance] 
I1217 21:17:53.596] test/e2e/apimachinery/resource_quota.go:112
I1217 21:17:53.596] 
I1217 21:17:53.596] Ran 802 of 5668 Specs in 1276.063 seconds
I1217 21:17:53.596] FAIL! -- 801 Passed | 1 Failed | 0 Pending | 4866 Skipped
```
/test pull-kubernetes-e2e-gce-ubuntu-containerd

deads2k commented on test/conformance/testdata/conformance.yaml:
```diff
@@ -659,7 +659,7 @@
   description: Ensure that the sample-apiserver code from 1.17 and compiled against
     1.17 will work on the current Aggregator/API-Server.
-  release: v1.17
+  release: v1.17, v1.21
```
what does this do?

johnbelamaric replied:
It just documents that the test was first introduced to conformance in 1.17 but that there was a change in behavior in 1.21

deads2k commented:
The code change looks reasonable, but I don't know what https://github.com/kubernetes/kubernetes/pull/97327/files#r552912579 means.

Riaankl commented:
@deads2k the test was promoted in 1.17 and brought create-, read- & deleteApiregistrationV1APIService to conformance.

@heyste now updated the test to include readApiregistrationV1APIServiceStatus, patchApiregistrationV1APIService, listApiregistrationV1APIService. (APISnoop [link](https://apisnoop.cncf.io/1.20.0/stable/apiregistration))

The requirements for [Conformance Test Comment Metadata](github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/conformance-tests.md#conformance-test-comment-metadata) when updating test in later releases:
" `Release`: indicates the Kubernetes release that the test was added to the conformance test suite. If the test was modified in subsequent releases then those releases should be included as well (comma separated) "

To ensure the code does not bring any surprises when merged, we did a lot of retesting and all seem well. 

If that clarifies your question can you please add a `/approve`

deads2k commented:
thanks.

/approve

Riaankl commented:
/test pull-kubernetes-bazel-test

Riaankl commented:
[Testgrid ](https://testgrid.k8s.io/sig-release-master-blocking#gce-cos-master-default&width=5&include-filter-by-regex=Sample%20API%20Server%20using%20the%20current%20Aggregator) is looking good after this PR merged.
